src/esc/commands.ts

```diff
@@ -129,6 +148,95 @@ export function openEnvironmentCommand(): vscode.Disposable {
     });
 }
 
+async function offerCreateOpenRequest(api: EscApi, org: string, project: string, envName: string) {
+    const selection = await vscode.window.showErrorMessage(
+        `Cannot open "${envName}". You must to request approval.`,
+        'Request Open Access',
+        'Cancel'
+    );
+
+    if (selection !== 'Request Open Access') {
+        return;
+    }
+
+    const description = await vscode.window.showInputBox({
+        prompt: 'Why are you requesting open access?',
+        placeHolder: ''
+    });
+    if (description === undefined) {
+        return;
+    }
+
+    const accessDuration = await vscode.window.showInputBox({
+        prompt: 'How long do you need to access the environment?',
+        placeHolder: 'e.g., 1h, 30m, 1d, 2h30m',
+        value: '1h',
+        validateInput: (value) => {
+            const seconds = parseDuration(value);
+            if (seconds === null) {
+                return inputError('Invalid duration format. Use d (days), h (hours), or m (minutes), e.g., 1h, 30m, 1d2h');
+            }
+            if (seconds <= 0) {
+                return inputError('Duration must be greater than 0');
+            }
+            return null;
+        }
+    });
+    if (accessDuration === undefined) {
+        return;
+    }
+
+    const durationSeconds = parseDuration(accessDuration);
+    if (durationSeconds === null) {
+        return;
+    }
+
+    const changeRequestId = await api.createOpenRequest(org, project, envName, durationSeconds);
+    await api.submitChangeRequest(changeRequestId, description);
+
+    const changeRequestUrl = `${config.consoleUrl()}/${org}/esc/${project}/${envName}/change-requests?requestId=${changeRequestId}`;
+    vscode.window.showInformationMessage(
+        'Access request created. Please wait for approval.',
+        'Open Access Request in Browser',
+    ).then(result => {
+        if (result === 'Open Access Request in Browser') {
+            vscode.env.openExternal(vscode.Uri.parse(changeRequestUrl));
+        }
+    });
```

Comment:
For consistency, we should use `await` instead of promise api

```suggestion
    const accessRequestPrompResult = await vscode.window.showInformationMessage(
        'Access request created. Please wait for approval.',
        'Open Access Request in Browser',
    );
    
     if (accessRequestPrompResult === 'Open Access Request in Browser') {
            vscode.env.openExternal(vscode.Uri.parse(changeRequestUrl));
       }
```